File: Proyectos Básicos/CalculadoraPy/CalculadoraSimple/CalculadoraSimple.py
def sumar(a, b):
    return a + b

def restar(a, b):
    return a - b

def multiplicar(a, b):
    return a * b

def dividir(a, b):
    if b == 0:
        return "Error: No se puede dividir por cero."
    return a / b

def calculadora():
    print("Bienvenidos a la Calculadora Simple")
    print("Selecciona una operación:")
    print("1. Sumar")
    print("2. Restar")
    print("3. Multiplicar")
    print("4. Dividir")

    while True:
        opcion = input("Ingrese el número de la operación (1/2/3/4) o 'q' para salir: ")

        if opcion.lower() == 'q':
            print("¡Gracias por usar la calculadora! Adiós.")
            break

        if opcion not in ['1', '2', '3', '4']:
            print("Opción inválida. Intenta de nuevo.")
            continue

        try:
            num1 = float(input("Ingresa el primer número: "))
            num2 = float(input("Ingresa el segundo número: "))
        except ValueError:
            print("Por favor, ingresa números válidos.")
            continue

        if opcion == '1':
            resultado = sumar(num1, num2)
            print(f"El resultado de {num1} + {num2} es: {resultado}")
        elif opcion == '2':
            resultado = restar(num1, num2)
            print(f"El resultado de {num1} - {num2} es: {resultado}")
        elif opcion == '3':
            resultado = multiplicar(num1, num2)
            print(f"El resultado de {num1} * {num2} es: {resultado}")
        elif opcion == '4':
            resultado = dividir(num1, num2)
            print(f"El resultado de {num1} / {num2} es: {resultado}")

        print("\n")

if __name__ == "__main__":
    calculadora()

File: Proyectos Básicos/CalculadoraJava/CalculadoraSimple/src/calculadorasimple/CalculadoraSimple.java
package calculadorasimple;

import java.util.Scanner;

public class CalculadoraSimple {

    public static double sumar(double a, double b) {
        return a + b;
    }

    public static double restar(double a, double b) {
        return a - b;
    }

    public static double multiplicar(double a, double b) {
        return a * b;
    }

    public static double dividir(double a, double b) {
        if (b == 0) {
            System.out.println("Error: No se puede dividir por cero.");
            return Double.NaN;
        }
        return a / b;
    }

    public static void main(String[] args) {
        Scanner entrada = new Scanner(System.in);

        System.out.println("Bienvenido a la Calculadora Simple");
        System.out.println("1. Sumar");
        System.out.println("2. Restar");
        System.out.println("3. Multiplicar");
        System.out.println("4. Dividir");
        System.out.println("5. Salir");

        while (true) {
            System.out.println("\nIngresa tu opción (1/2/3/4/5): ");
            int opcion = entrada.nextInt();

            if (opcion == 5) {
                System.out.println("¡Gracias por usar la calculadora! Adiós.");
                break;
            }

            System.out.println("Ingrese el primer número: ");
            double num1 = entrada.nextDouble();
            System.out.println("Ingresa el segundo número: ");
            double num2 = entrada.nextDouble();

            double resultado;
            switch (opcion) {
                case 1:
                    resultado = sumar(num1, num2);
                    System.out.println("El resultado de la suma es: " + resultado);
                    break;
                case 2:
                    resultado = restar(num1, num2);
                    System.out.println("El resultado de la resta es: " + resultado);
                    break;
                case 3:
                    resultado = multiplicar(num1, num2);
                    System.out.println("El resultado de la multiplicación es: " + resultado);
                    break;
                case 4:
                    resultado = dividir(num1, num2);
                    if (!Double.isNaN(resultado)) {
                        System.out.println("El resultado de la división es: " + resultado);
                    }
                    break;
                default:
                    System.out.println("Opción no válida. Intente de nuevo.");
            }
        }
        entrada.close();
    }
}
